### PR TITLE
Lookup button enabled/disabled depending on Lookup Dictionary preference 

### LINKED ISSUE #7679: Reviewer: "Lookup Dictionary" menu item is unused
Functionality is hooked up, but not usable as the control is never made visible.

We should add this to "App Bar Buttons" so the user can move this around on the menu.

https://github.com/ankidroid/Anki-Android/blob/46075e2ccc93a40e3b64c39333263184f43fbe07/AnkiDroid/src/main/res/menu/reviewer.xml#L118-L121

### PR DESCRIPTION
**Pull request by Mohaned-Elfeky**
## Purpose / Description
Made the select text Action Button visible/invisible depending on the Look Up dictionary preference and made the button display a toast if no text was selected.

## Fixes
Fixes #7679

## Checklist
_Please, go through these checks before submitting the PR._

- [ x ] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ x ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ x ] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ x ] You have commented your code, particularly in hard-to-understand areas
- [ x ] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java [hunk 1]
```diff
@@ -1339,7 +1340,7 @@ protected void lookUpOrSelectText() {
             Timber.d("Clipboard has text = %b", clipboardHasText());
             lookUp();
         } else {
-            selectAndCopyText();
+            UIUtils.showThemedToast(AbstractFlashcardViewer.this, "Clipboard is empty" , false);
```
Comment (4D17Y4) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:1343]: Use string from resources instead .

File: AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ActionButtonStatus.java [hunk 2]
```diff
@@ -65,6 +66,13 @@ public void setup(SharedPreferences preferences) {
         setupButton(preferences, R.id.action_toggle_whiteboard, "customButtonEnableWhiteboard", SHOW_AS_ACTION_NEVER);
         setupButton(preferences, R.id.action_save_whiteboard, "customButtonSaveWhiteboard", SHOW_AS_ACTION_NEVER);
         setupButton(preferences, R.id.action_change_whiteboard_pen_color, "customButtonWhiteboardPenColor", SHOW_AS_ACTION_IF_ROOM);
+        if(Lookup.isAvailable()){  //checks if Lookup Dictionary preference is enabled
+            setupButton(preferences, R.id.action_search_dictionary, "customButtonSearchDictionary",SHOW_AS_ACTION_IF_ROOM);
+        }
+        else{
+            setupButton(preferences, R.id.action_search_dictionary, "customButtonSearchDictionary",MENU_DISABLED);
+        }
```
Comment (4D17Y4) [inline AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ActionButtonStatus.java:74]: Spacing according to our code style: https://github.com/ankidroid/Anki-Android/wiki/Code-style

File: AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java [hunk 3]
```diff
@@ -7,6 +7,7 @@
 import android.content.SharedPreferences;
 import android.net.Uri;
 
+
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java:10]: Could you revert unnecessary changes to files

File: AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ActionButtonStatus.java (hunk 2 above)
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ActionButtonStatus.java:74]: Even with the correction, this isn't correct. We want to disable the button if lookup is disabled, but the final parameter is the default if the preference isn't set, so the button could still appear, even if lookup is disabled

```suggestion
         setupButton(preferences, R.id.action_search_dictionary, "customButtonSearchDictionary", Lookup.isAvailable() ? SHOW_AS_ACTION_IF_ROOM : MENU_DISABLED);
```

**Comment by mrudultora** [comment]
Hi @mohaned-hassan thanks for the PR. Could you please change your PR title to something more understable? Giving a title **My feature branch** shows nothing about the changes you made. Also, rather than `Issue #7679` in the Fixes section, write `Fixes #7679`. This will link that issue with this PR.

**Review by 4D17Y4 — COMMENTED** [review]
Minor changes

File: AnkiDroid/src/main/res/xml/preferences_custom_buttons.xml [hunk 4]
```diff
@@ -102,6 +102,12 @@ TODO: Add a unit test
             android:entryValues="@array/custom_button_values"
             android:key="customButtonToggleMicToolBar"
             android:title="@string/menu_toggle_mic_tool_bar" />
+        <ListPreference
+            android:defaultValue="0"
+            android:entries="@array/custom_button_labels"
+            android:entryValues="@array/custom_button_values"
+            android:key="customButtonSearchDictionary"
+            android:title="@string/menu_select" />
```
Comment (david-allison-1) [inline AnkiDroid/src/main/res/xml/preferences_custom_buttons.xml:110]: `Select text` won't mean anything to a user,

**Review by david-allison-1 — COMMENTED** [review]
Thanks!

File: AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ActionButtonStatus.java (hunk 2 above)
Comment (mohaned-hassan) [reply AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ActionButtonStatus.java:74]: can you please suggest a fix for this ?

**Comment by david-allison-1** [comment]
@Mohaned-Elfeky Can this be closed if you've opened a second PR?

**Comment by mikehardy** [comment]
Superceded by #8441
